### PR TITLE
jupyter: new image with 4 new extensions

### PR DESCRIPTION
**Pull request by ChaamC**
The google drive extension for JupyterLab requires a settings file containing the clientid of the project created in developers.google.com, which give authorization to use google drive.

This PR's role is to include this file in the birdhouse configs.

Matching PR https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/pull/54 (commit https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/commit/5d5a9aa2251386378406efb5b414b3aa6db0b37e) for the new image with 4 new extensions: jupytext, jupyterlab-google-drive, jupyter_conda and jupyterlab-git

Matching PR https://github.com/Ouranosinc/pavics-sdi/pull/185 for documention about the new extensions.

### CODE REVIEW

**Review by dbyrns — APPROVED** [review]

**Review by MatProv — APPROVED** [review]

**Review by tlvu — CHANGES_REQUESTED** [review]
Thanks for the changes.  I'd like to be defensive in case this new option is not set and I'd like to have a build with it that I can go in and try to use it, is that possible with the pipeline?  I not I'll try this manually on my side.

**Comment by tlvu** [comment]
Matching PR https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/pull/54 that introduced the jupyterlab-google-drive extension needing this change.

**Comment by MatProv** [comment]
run tests

File: birdhouse/config/jupyterhub/jupyterhub_config.py.template [hunk 1]
```diff
@@ -37,6 +37,14 @@ c.DockerSpawner.environment = {
 
 c.DockerSpawner.volumes = {host_user_data_dir: container_workspace_dir}
 
+container_gdrive_settings_path = join(container_home_dir, ".jupyter/lab/user-settings/@jupyterlab/google-drive/drive.jupyterlab-settings")
+host_gdrive_settings_path = os.environ['JUPYTER_GOOGLE_DRIVE_SETTINGS']
+
+c.DockerSpawner.volumes[host_gdrive_settings_path] = {
```
[the remark this answers is not in the document]
Comment (ChaamC) [reply birdhouse/config/jupyterhub/jupyterhub_config.py.template]: I added the proposed fix for the empty settings case!

There was no build, cause I think the build were only triggered for certain users in a whitelist. @MatProv fixed this.

We can use a different settings file depending of the environment. We will have our own settings file for 'outarde' for example, and you could have another settings file for ouranos, with a clientId for your own google project. The settings file is defined in env.local.

**Comment by MatProv** [comment]
run tests

**Comment by tlvu** [comment]
run tests

**Review by tlvu — APPROVED** [review]

**Pull request merged**